src/main/java/com/fasterxml/jackson/databind/deser/std/PrimitiveArrayDeserializers.java

```diff
@@ -81,6 +83,9 @@ protected PrimitiveArrayDeserializers(PrimitiveArrayDeserializers<?> base,
     public JsonDeserializer<?> createContextual(DeserializationContext ctxt,
             BeanProperty property) throws JsonMappingException
     {
+    	if(property != null) {
+    		_format =  property.findPropertyFormat(ctxt.getConfig(), property.getType().getRawClass());
```

Comment:
This can not be done, as the intent with `createContextual` is to never modify instance but to create a new differently constructed instance if necessary. This because serializers/deserializers are shared by all properties of the same type, so any per-property annotations must only take effect for serializers/deserializers used for them.

So, the pattern to use is to collect all pertinent information, and create & return new instance if necessary based on different configuration.